### PR TITLE
chore(examples): Fix functions examples + add google oauth

### PR DESCRIPTION
**Pull request by graysonhicks**
This creates an example for logging in and out with Google OAuth and Gatsby Functions.  It also renames an existing 'google-auth' example to 'google-gmail' to be more accurate in the difference between the two.

This example will compliment a blog post that is in review.

### CODE REVIEW

File: examples/functions-google-gmail/README.md [hunk 1]
```diff
@@ -31,7 +31,7 @@
     When developing locally, you include environment variables to your `.env.development`. Read more about how Gatsby handles `.env` files and environment variables in the [Gatsby Docs](https://www.gatsbyjs.com/docs/how-to/local-development/environment-variables/)
 
     ```shell
-    cd google-auth
+    cd google-gmail
```
Comment (LekoArts) [inline examples/functions-google-gmail/README.md:34]: But that's not the name of the directory? If we want to be super correct it has to be `cd examples/functions-google-gmail`

File: examples/functions-google-oauth/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    There are **2** environment variable you'll need to add to your project:
+
+    - `GOOGLE_CLIENT_ID`: Fetch the ClientId from the newly created OAuth 2.0 Client ID. Yuo can find it in the [Google Credentials Console](https://console.cloud.google.com/apis/credentials)
+    - `GOOGLE_CLIENT_SECRET`: Fetch the Client Secret from the newly created OAuth 2.0 Client ID. Yuo can find it in the [Google Credentials Console](https://console.cloud.google.com/apis/credentials)
+
+    You'll also want to add these as environment variables when deploying to Gatsby Cloud.
+
+3.  **Start developing.**
+
+    To get started, run `yarn` to add all necessary packages.
```
Comment (LekoArts) [inline examples/functions-google-oauth/README.md:28]: Unfortunately the already merged examples used `yarn` in their examples. This should be fixed in a follow-up.

For this PR we can do it right before the merge. So any instance of `yarn` should be replaced with `npm`

File: examples/functions-google-oauth/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    When developing locally, you include environment variables to your `.env.development`. Read more about how Gatsby handles `.env` files and environment variables in the [Gatsby Docs](https://www.gatsbyjs.com/docs/how-to/local-development/environment-variables/)
+
+    ```shell
+    cd functions-google-oauth
```
Comment (LekoArts) [inline examples/functions-google-oauth/README.md:33]: Same comment as in other example above

File: examples/functions-google-oauth/gatsby-config.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+module.exports = {
+  siteMetadata: {
+    title: "Gatsby Functions with Google OAuth Example",
+  },
+  // Only need for < 3.7.0 Gatsby
```
Comment (LekoArts) [inline examples/functions-google-oauth/gatsby-config.js:5]: Let's remove it, the example supports it so no need to add it here

File: examples/functions-google-oauth/gatsby-config.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  // flags: {
+  //   FUNCTIONS: true,
+  // },
+  plugins: [
+    "gatsby-plugin-gatsby-cloud",
+    `gatsby-plugin-sass`,
+    {
+      resolve: `gatsby-plugin-create-client-paths`,
```
Comment (LekoArts) [inline examples/functions-google-oauth/gatsby-config.js:13]: We shouldn't use this plugin and rather use the File System Route API

File: examples/functions-google-oauth/src/api/googleAccessToken.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+// Intialize our Google OAuth client, same as in login function.
+const oauth2Client = new google.auth.OAuth2(
+  clientId,
+  clientSecret,
+  process.env.GOOGLE_REDIRECT_URI
+)
+
+google.options({ auth: oauth2Client })
+
+export default async (req, res) => {
```
Comment (LekoArts) [inline examples/functions-google-oauth/src/api/googleAccessToken.js:15]: Anonymous functions are bad practice (e.g. for debugging)

File: examples/functions-google-oauth/src/api/login.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  // Redirect URL:
+  // http://localhost:8000/api/googleAccessToken
+  process.env.GOOGLE_REDIRECT_URI
+)
+
+google.options({ auth: oauth2Client })
+
+const scopes = ["profile"]
+
+export default async (req, res) => {
```
Comment (LekoArts) [inline examples/functions-google-oauth/src/api/login.js:19]: anonymous

**Review by LekoArts — CHANGES_REQUESTED** [review]
Thanks!

**Comment by graysonhicks** [comment]
Thanks @LekoArts these should be resolved now!

**Review by LekoArts — APPROVED** [review]

**Pull request merged**